src/games/strategy/util/Triple.java

```diff
@@ -14,7 +14,7 @@
     return new Triple(first,second,third);
   }
 
-  private Triple(final F first, final S second, final T third) {
+  public Triple(final F first, final S second, final T third) {
```

Comment:
Please leave this private, use the static method "Triple.of" instead.
I'd recommend not using "Triple" in favor of creating your own simple data class. The metthods "getFirst()" ,"getSecond", etc are not easy to understand from a readers point of view.